#!/bin/sh /usr/share/dpatch/dpatch-run
## 02-module-in-kernel.dpatch by Michael Meskes <meskes@debian.org>
##
## DP: Make sure modules build in a kernel build environment.

@DPATCH@

--- virtualbox-ose-1.5.4-dfsg/src/VBox/Additions/linux/sharedfolders/Makefile.module.orig	2007-09-24 22:13:56.000000000 +0200
+++ virtualbox-ose-1.5.4-dfsg/src/VBox/Additions/linux/sharedfolders/Makefile.module	2008-01-01 13:03:53.000000000 +0100
@@ -38,28 +38,30 @@
 
 ifneq ($(MAKECMDGOALS),clean)
 
-# kernel base directory
-ifndef KERN_DIR
- KERN_DIR := /lib/modules/$(shell uname -r)/build
- ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
-  KERN_DIR := /usr/src/linux
+ifeq ($(KERNELRELEASE),)
+ # kernel base directory
+ ifndef KERN_DIR
+  KERN_DIR := /lib/modules/$(shell uname -r)/build
   ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
-   $(error Error: unable to find the sources of your current Linux kernel. Specify KERN_DIR=<directory> and run Make again.)
+   KERN_DIR := /usr/src/linux
+   ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
+    $(error Error: unable to find the sources of your current Linux kernel. Specify KERN_DIR=<directory> and run Make again.)
+   endif
+   $(warning Warning: using /usr/src/linux as the source directory of your Linux kernel. If this is not correct, specify KERN_DIR=<directory> and run Make again.)
+  endif
+ else
+  ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
+   $(error Error: KERN_DIR does not point to a directory.)
   endif
-  $(warning Warning: using /usr/src/linux as the source directory of your Linux kernel. If this is not correct, specify KERN_DIR=<directory> and run Make again.)
- endif
-else
- ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
-  $(error Error: KERN_DIR does not point to a directory.)
  endif
-endif
 
-# includes
-ifndef KERN_INCL
- KERN_INCL = $(KERN_DIR)/include
-endif
-ifneq ($(shell if test -d $(KERN_INCL); then echo yes; fi),yes)
- $(error Error: unable to find the include directory for your current Linux kernel. Specify KERN_INCL=<directory> and run Make again.)
+ # includes
+ ifndef KERN_INCL
+  KERN_INCL = $(KERN_DIR)/include
+ endif
+ ifneq ($(shell if test -d $(KERN_INCL); then echo yes; fi),yes)
+  $(error Error: unable to find the include directory for your current Linux kernel. Specify KERN_INCL=<directory> and run Make again.)
+ endif
 endif
 
 # module install dir.
@@ -95,7 +97,10 @@
 # Compiler options
 #
 ifndef INCL
- INCL    := -I$(KERN_INCL) $(addprefix -I, $(EXTRA_INCL))
+ ifeq ($(KERNELRELEASE),)
+  INCL    := -I$(KERN_INCL)
+ endif
+ INCL += $(addprefix -I, $(EXTRA_INCL))
  ifndef KBUILD_EXTMOD
   KBUILD_EXTMOD := $(shell pwd)
  endif
--- virtualbox-ose-1.5.4-dfsg/src/VBox/HostDrivers/Support/linux/Makefile.orig	2008-01-01 12:58:40.000000000 +0100
+++ virtualbox-ose-1.5.4-dfsg/src/VBox/HostDrivers/Support/linux/Makefile	2008-01-01 13:00:24.000000000 +0100
@@ -58,37 +58,39 @@
 
 ifneq ($(MAKECMDGOALS),clean)
 
-# kernel base directory
-ifndef KERN_DIR
- # build for the current kernel, version check
- KERN_DIR := /lib/modules/$(shell uname -r)/build
- ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
-  KERN_DIR := /usr/src/linux
+ifeq ($(KERNELRELEASE),)
+ # kernel base directory
+ ifndef KERN_DIR
+  # build for the current kernel, version check
+  KERN_DIR := /lib/modules/$(shell uname -r)/build
   ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
-   $(error Error: unable to find the sources of your current Linux kernel. Specify KERN_DIR=<directory> and run Make again.)
+   KERN_DIR := /usr/src/linux
+   ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
+    $(error Error: unable to find the sources of your current Linux kernel. Specify KERN_DIR=<directory> and run Make again.)
+   endif
+   $(warning Warning: using /usr/src/linux as the source directory of your Linux kernel. If this is not correct, specify KERN_DIR=<directory> and run Make again.)
   endif
-  $(warning Warning: using /usr/src/linux as the source directory of your Linux kernel. If this is not correct, specify KERN_DIR=<directory> and run Make again.)
- endif
- # check if versions match -- works only for later 2.6 kernels
- VBOX_KERN_VER := $(shell $(MAKE) -sC $(KERN_DIR) kernelrelease 2> /dev/null || true)
- ifneq ($(VBOX_KERN_VER),)
-  ifneq ($(VBOX_KERN_VER),$(shell uname -r))
-   $(error Error: /usr/src/linux (version $(VBOX_KERN_VER)) does not match the current kernel (version $(shell uname -r)))
+  # check if versions match -- works only for later 2.6 kernels
+  VBOX_KERN_VER := $(shell $(MAKE) -sC $(KERN_DIR) kernelrelease 2> /dev/null || true)
+  ifneq ($(VBOX_KERN_VER),)
+   ifneq ($(VBOX_KERN_VER),$(shell uname -r))
+    $(error Error: /usr/src/linux (version $(VBOX_KERN_VER)) does not match the current kernel (version $(shell uname -r)))
+   endif
+  endif
+ else
+  # build for a dedicated kernel, no version check
+  ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
+   $(error Error: KERN_DIR does not point to a directory.)
   endif
  endif
-else
- # build for a dedicated kernel, no version check
- ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
-  $(error Error: KERN_DIR does not point to a directory.)
- endif
-endif
 
-# includes
-ifndef KERN_INCL
- KERN_INCL = $(KERN_DIR)/include
-endif
-ifneq ($(shell if test -d $(KERN_INCL); then echo yes; fi),yes)
- $(error Error: unable to find the include directory for your current Linux kernel. Specify KERN_INCL=<directory> and run Make again.)
+ # includes
+ ifndef KERN_INCL
+  KERN_INCL = $(KERN_DIR)/include
+ endif
+ ifneq ($(shell if test -d $(KERN_INCL); then echo yes; fi),yes)
+  $(error Error: unable to find the include directory for your current Linux kernel. Specify KERN_INCL=<directory> and run Make again.)
+ endif
 endif
 
 # module install dir, only for current kernel
@@ -118,7 +120,10 @@
 # Compiler options
 #
 ifndef INCL
- INCL    := -I$(KERN_INCL) $(addprefix -I, $(EXTRA_INCL))
+ ifeq ($(KERNELRELEASE),)
+  INCL    := -I$(KERN_INCL)
+ endif
+ INCL += $(addprefix -I, $(EXTRA_INCL))
  ifndef KBUILD_EXTMOD
   KBUILD_EXTMOD := $(shell pwd)
  endif
--- virtualbox-ose-1.5.4-dfsg/src/VBox/Additions/linux/module/Makefile.module.orig	2008-01-01 13:19:09.000000000 +0100
+++ virtualbox-ose-1.5.4-dfsg/src/VBox/Additions/linux/module/Makefile.module	2008-01-01 13:20:44.000000000 +0100
@@ -31,28 +31,30 @@
 
 ifneq ($(MAKECMDGOALS),clean)
 
-# kernel base directory
-ifndef KERN_DIR
- KERN_DIR := /lib/modules/$(shell uname -r)/build
- ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
-  KERN_DIR := /usr/src/linux
+ifeq ($(KERNELRELEASE),)
+ # kernel base directory
+ ifndef KERN_DIR
+  KERN_DIR := /lib/modules/$(shell uname -r)/build
   ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
-   $(error Error: unable to find the sources of your current Linux kernel. Specify KERN_DIR=<directory> and run Make again.)
+   KERN_DIR := /usr/src/linux
+   ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
+    $(error Error: unable to find the sources of your current Linux kernel. Specify KERN_DIR=<directory> and run Make again.)
+   endif
+   $(warning Warning: using /usr/src/linux as the source directory of your Linux kernel. If this is not correct, specify KERN_DIR=<directory> and run Make again.)
+  endif
+ else
+  ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
+   $(error Error: KERN_DIR does not point to a directory.)
   endif
-  $(warning Warning: using /usr/src/linux as the source directory of your Linux kernel. If this is not correct, specify KERN_DIR=<directory> and run Make again.)
- endif
-else
- ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
-  $(error Error: KERN_DIR does not point to a directory.)
  endif
-endif
 
-# includes
-ifndef KERN_INCL
- KERN_INCL = $(KERN_DIR)/include
-endif
-ifneq ($(shell if test -d $(KERN_INCL); then echo yes; fi),yes)
- $(error Error: unable to find the include directory for your current Linux kernel. Specify KERN_INCL=<directory> and run Make again.)
+ # includes
+ ifndef KERN_INCL
+  KERN_INCL = $(KERN_DIR)/include
+ endif
+ ifneq ($(shell if test -d $(KERN_INCL); then echo yes; fi),yes)
+  $(error Error: unable to find the include directory for your current Linux kernel. Specify KERN_INCL=<directory> and run Make again.)
+ endif
 endif
 
 # module install dir.
@@ -88,7 +90,10 @@
 # Compiler options
 #
 ifndef INCL
- INCL    := -I$(KERN_INCL) $(addprefix -I, $(EXTRA_INCL))
+ ifeq ($(KERNELRELEASE),)
+  INCL    := -I$(KERN_INCL)
+ endif
+ INCL += $(addprefix -I, $(EXTRA_INCL))
  ifndef KBUILD_EXTMOD
   KBUILD_EXTMOD := $(shell pwd)
  endif
